libfreerdp/core/peer.c

```diff
@@ -735,7 +735,7 @@ static BOOL freerdp_peer_close(freerdp_peer* client)
 static void freerdp_peer_disconnect(freerdp_peer* client)
 {
 	rdpTransport* transport = client->context->rdp->transport;
-	transport_disconnect(transport);
+	client->context->update->io->TransportDisconnect(transport);
```

Comment:
I don´t like the exposure of the internal structure of `rdpTransport` you´re doing here.
We are working on moving all module/components to be self contained with only an opaque structure exposed, this moves in the opposite direction.